mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -126,6 +131,87 @@ where
 {
 }
 
+#[derive(Deserialize)]
+#[serde(default)]
+pub struct MergeConditions {
+    /// The minimum number of UTXOs to merge. If the number of UTXOs is less than this, the merge will not be performed.
+    pub merge_at: usize,
+    /// The maximum number of UTXOs to merge at once in a single transaction.
+    pub max_merge_at_once: usize,
+}
+
+impl Default for MergeConditions {
+    fn default() -> Self {
+        MergeConditions {
+            merge_at: 50,
+            max_merge_at_once: 50,
+        }
+    }
+}
+
+/// Merges unspent UTXOs from `from_address` address to `to_script_pubkey` script.
+pub async fn merge_utxos<Coin>(
+    coin: &Coin,
+    from_address: &Address,
+    to_script_pubkey: &Script,
+    merge_conditions: &MergeConditions,
+    broadcast: bool,
+) -> MmResult<(Transaction, Vec<UnspentInfo>), String>
+where
+    Coin: UtxoCommonOps + GetUtxoListOps,
+{
+    let ticker = &coin.as_ref().conf.ticker;
+    let (unspents, recently_spent) = coin
+        .get_unspent_ordered_list(from_address)
+        .await
+        .mm_err(|e| format!("Error in get_unspent_ordered_list for coin={ticker}: {e}"))?;
+
+    if unspents.len() < merge_conditions.merge_at {
+        return Err(format!(
+            "Not enough unspent UTXOs to merge for coin={ticker}, found={}, required={}",
+            unspents.len(),
+            merge_conditions.merge_at
+        )
+        .into());
+    }
+    let unspents: Vec<_> = unspents.into_iter().take(merge_conditions.max_merge_at_once).collect();
+    if unspents.len() < 2 {
+        return Err(format!("No point of merging only a single UTXO (coin={ticker})").into());
+    }
+
+    let value = unspents.iter().fold(0, |sum, unspent| sum + unspent.value);
+    let output = TransactionOutput {
+        value,
+        script_pubkey: to_script_pubkey.to_bytes(),
+    };
+
+    let tx = if broadcast {
+        generate_and_send_tx(
+            coin,
+            unspents.clone(),
+            None,
+            FeePolicy::DeductFromOutput(0),
```

Comment:
It's unrelated to this PR but `DeductFromOutput` index doesn't really make sense here :)
I guess this needs a refactor later.